spec/unit/state-lookup.spec.js

```diff
@@ -157,18 +158,15 @@ describe("StateLookup", function() {
             var p = lookup.trackRoom("!foo:bar");
             expect(p.isPending()).toBe(true); // not resolved HTTP call yet
             // this event should clobber response from HTTP call
-            lookup.onEvent(
+            statePromise.resolve();
+            await lookup.onEvent(
                 {type: "m.room.name", state_key: "", room_id: "!foo:bar",
                     content: { name: "Bar" }}
             );
-            statePromise.resolve();
-
-            p.then(function() {
-                expect(
-                    lookup.getState("!foo:bar", "m.room.name", "").content.name
-                ).toEqual("Bar");
-                done();
-            });
+            await p;
+            expect(
+                lookup.getState("!foo:bar", "m.room.name", "").content.name
+            ).toEqual("Bar");
```

Comment:
Let's take a minute and appreciate how much cleaner this code looks now. 👏
Also, on an error, the test will fail instead of time out. 👏👏

Reply:
:)